examples/Org.Website.Samples/TopicSample.cpp

```diff
@@ -17,20 +17,18 @@
 
 using namespace hazelcast::client;
 
-class TopicSample {
-public:
-    void operator()(topic::Message &&message) {
-        std::cout << "Got message " << message.getMessageObject().get<std::string>().value_or("null") << std::endl;
-    }
-};
-
 int main() {
     // Start the Hazelcast Client and connect to an already running Hazelcast Cluster on 127.0.0.1
     HazelcastClient hz;
     // Get a Topic called "my-distributed-topic"
     auto topic = hz.getTopic("my-distributed-topic");
     // Add a Listener to the Topic
-    topic->addMessageListener(TopicSample{}).get();
+    topic->addMessageListener(
+        topic::MessageListener().
+            on_received([](const topic::Message &message) {
```

Comment:
previous API is passing as rvalue and I think we can keep it that way, since we will discard the object, better yet just pass it to the user so that he/she can move around without copy.

Reply:
But then it is not consistent with the other listeners, can we pass EntryEvent, ItemEvent etc. as rvalue?

Reply:
We could, we are currently composing a separate event for each event message for each listener. If we are to make the event shared between listeners in the future, then it may be a different story but sharing events between listeners currently can only be considered for local listeners such as membership listener and such but for data structure listeners, the current way is a separate message for each listener even if they registered to the same data structure with the same flags.

Reply:
Ok, converted parameters back to `Message &&`. Created an issue for converting other events: https://github.com/hazelcast/hazelcast-cpp-client/issues/610